### DISCUSSION TITLE
Touch screen does not work on haspmota with sunton EPS32-8048S050

### DISCUSSION THREAD

**Discussion by Be-Virtual** [arendst/Tasmota#18565] [Q&A] [answered]
### PROBLEM DESCRIPTION
Touch screen does not work on haspmota with Sunton EPS32-8048S050 5 inch screen
Works OK on openhasp
Changing page with basic demo (3 button) manually also work ok.
Suppose autoconf file has an error but was not able to find description on config file on how to change it

### REQUESTED INFORMATION
_Make sure your have performed every step and checked the applicable boxes before submitting your issue. Thank you!_

- [X] Read the [Contributing Guide and Policy](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md) and [the Code of Conduct](https://github.com/arendst/Tasmota/blob/development/CODE_OF_CONDUCT.md)
- [X] Searched the problem in [issues](https://github.com/arendst/Tasmota/issues)
- [X] Searched the problem in [discussions](https://github.com/arendst/Tasmota/discussions)
- [X] Searched the problem in the [docs](https://tasmota.github.io/docs/FAQ)
- [X] Searched the problem in the [chat](https://discord.gg/Ks2Kzd4)
- [X] Device used (e.g., Sonoff Basic): Sunton ESP32-8048S050
- [X] Tasmota binary firmware version number used: 12.5.0.1
  - [X] Pre-compiled
  - [ ] Self-compiled
- [X] Flashing tools used: web
- [ ] Provide the output of command: `Backlog Template; Module; GPIO 255`:
```lua
  Configuration output here:


```
- [ ] If using rules, provide the output of this command: `Backlog Rule1; Rule2; Rule3`:
```lua
  Rules output here:


```
- [ ] Provide the output of this command: `Status 0`:
```lua
  STATUS 0 output here:


```
- [ ] Set `weblog` to 4 and then, when you experience your issue, provide the output of the Console log:
```lua
  Console output here:


```

### TO REPRODUCE
_Steps to reproduce the behavior: install on sunton devices

### EXPECTED BEHAVIOUR
_A clear and concise description of what you expected to happen. correct autoconf file (I suppose)

### SCREENSHOTS
_If applicable, add screenshots to help explain your problem._

### ADDITIONAL CONTEXT
_Add any other context about the problem here._

**(Please, remember to close the issue when the problem has been addressed)**

**Comment by s-hadinger** [comment]
The touchscreen config is mostly in `display.ini`. Can you share your file?

Can you also share which basic demo you tried and which `pages.jsonl` failed?

**Reply by Be-Virtual** [reply]
> The touchscreen config is mostly in `display.ini`. Can you share your file?
> 
> Can you also share which basic demo you tried and which `pages.jsonl` failed?


Hi,

------------------ OPENHASP ----------------------------------

I installed the nightly openhasp to test if my screen was ok
Basic install from nightly, then used this :
// ------------------------------------- Screen 480 X 320 w=(159, 160, 159) -------------------------------------
{"page":0,"id":1,"obj":"btn","action":{"down": "page prev"},"x":0,"y":290,"w":159,"h":32,"bg_color":"#2C3E50","text":"\uE141","text_color":"#FFFFFF","radius":0,"border_side":0,"text_font":24}
{"page":0,"id":2,"obj":"btn","action":{"down": "page back"},"x":160,"y":290,"w":160,"h":32,"bg_color":"#2C3E50","text":"\uE2DC","text_color":"#FFFFFF","radius":0,"border_side":0,"text_font":24}
{"page":0,"id":3,"obj":"btn","action":{"down": "page next"},"x":321,"y":290,"w":159,"h":32,"bg_color":"#2C3E50","text":"\uE142","text_color":"#FFFFFF","radius":0,"border_side":0,"text_font":24}
{"page":1,"comment":" ---- Basic Page 1 ----"}

{"page":1,"id":10,"w":100,"obj":"label","text": "%hostname%","align":0,"text_font":16,"text_color":"white"}
{"page":1,"id":11,"obj":"label","x":100,"y":0,"w":100,"h":35,"text":"%ssid%","align":1,"text_font":16,"text_color":"white"}
{"page":1,"id":12,"obj":"label","x":200,"y":0,"w":120,"h":35,"text":"%ip%","align":2,"text_font":16,"text_color":"white"}

I can switch from screen with navigation keys on screen

------------------ TASMOTA / HASPMOTA ----------------------------------

Then I installed Tasmota with unofficial firmware 'Tasmota32-S3 (QIO Flash/OPI PSRAM)

I used the autoconf to set board to Senton 8048S050.
Content of Display.ini (from SunTon-S3-Display.autoconf):
:H,ST7262,800,480,16,RGB,40,41,39,42,2,15,16,4,45,48,47,21,14,8,3,46,9,1,5,6,7,14
:S,2,1,1,0,40,20
:V,0,8,4,8,0,8,4,8,1
:0,00
:1,01
:2,02
:3,03
:TI1,5d,,,-1,38

Content of init.bat (from SunTon-S3-Display.autoconf):
Template {"NAME":"SUNTON","GPIO":[1,1,1,1,1,1,1,1,1,1,6720,704,736,672,1,1,1,1,1,640,608,1,0,0,0,0,0,6210,1,1,1,1,1,1,1,1,1,1],"FLAG":0,"BASE":1}
Module 0
DisplayDimmer 70

I tested with the demo:
UrlFetch https://raw.githubusercontent.com/arendst/Tasmota/development/tasmota/berry/haspmota/haspmota_demo.tapp

and I tested with this code (partially from demo code):

{"page":0,"comment":"---------- Upper stat line ----------"}
{"page":0,"id":0,"text_color":"#FFFFFF"}
{"page":0,"id":11,"obj":"label","x":0,"y":0,"w":800,"pad_right":90,"h":22,"bg_color":"#D00000","bg_opa":255,"radius":0,"border_side":0,"text":"Tasmota","text_font":"montserrat-20"}

{"page":0,"id":15,"obj":"lv_wifi_arcs","x":771,"y":0,"w":29,"h":22,"radius":0,"border_side":0,"bg_color":"#000000","line_color":"#FFFFFF"}
{"page":0,"id":16,"obj":"lv_clock","x":712,"y":3,"w":55,"h":16,"radius":0,"border_side":0}

{"comment":"---------- Bottom buttons - prev/home/next ----------"}
{"page":0,"id":101,"obj":"btn","x":20,"y":450,"w":100,"h":25,"action":"prev","bg_color":"#1fa3ec","radius":10,"border_side":1,"text":"\uF053","text_font":"montserrat-20"}
{"page":0,"id":102,"obj":"btn","x":300,"y":450,"w":200,"h":25,"action":"back","bg_color":"#1fa3ec","radius":10,"border_side":1,"text":"\uF015","text_font":"montserrat-20"}
{"page":0,"id":103,"obj":"btn","x":680,"y":450,"w":100,"h":25,"action":"next","bg_color":"#1fa3ec","radius":10,"border_side":1,"text":"\uF054","text_font":"montserrat-20"}

{"page":1,"comment":"---------- Page 1 ----------"}
{"page":1,"id":0,"bg_color":"#0000A0","bg_grad_color":"#000000","bg_grad_dir":1,"text_color":"#FFFFFF"}

{"page":1,"id":2,"obj":"arc","x":20,"y":65,"w":80,"h":100,"border_side":0,"type":0,"rotation":0,"start_angle":180,"end_angle":0,"start_angle1":180,"value_font":12,"value_ofs_x":0,"value_ofs_y":-14,"bg_opa":0,"text":"--.-°C","min":200,"max":800,"val":0,"val_rule":"ESP32#Temperature","val_rule_formula":"val * 10","text_rule":"ESP32#Temperature","text_rule_format":"%2.1f °C"}

{"page":1,"id":5,"obj":"label","x":2,"y":35,"w":120,"text":"Temperature","align":1}

{"page":1,"id":10,"obj":"label","x":172,"y":35,"w":140,"text":"MPU","align":0}
{"page":1,"id":11,"obj":"label","x":172,"y":55,"w":140,"text":"x=","align":0,"text_rule":"MPU9250#AX","text_rule_format":"x=%6.3f","text_rule_formula":"val / 1000"}
{"page":1,"id":12,"obj":"label","x":172,"y":75,"w":140,"text":"y=","align":0,"text_rule":"MPU9250#AY","text_rule_format":"y=%6.3f","text_rule_formula":"val / 1000"}
{"page":1,"id":13,"obj":"label","x":172,"y":95,"w":140,"text":"z=","align":0,"text_rule":"MPU9250#AZ","text_rule_format":"z=%6.3f","text_rule_formula":"val / 1000"}

{"page":2,"comment":"---------- Page 2 ----------"}
{"page":2,"id":0,"bg_color":"#0000A0","bg_grad_color":"#000000","bg_grad_dir":1,"text_color":"#FFFFFF"}

{"comment":"---------- Wifi status ----------"}
{"page":2,"id":20,"obj":"lv_wifi_graph","x":700,"y":25,"w":90,"h":40,"radius":0}
{"page":2,"id":21,"obj":"lv_tasmota_info","x":3,"y":25,"w":695,"h":40,"radius":0}
{"page":2,"id":22,"obj":"lv_tasmota_log","x":3,"y":68,"w":790,"h":200,"radius":0,"text_font":12}

MPU9250 is not present on this board, so value were 0

If I go to the berry console i can switch pages with p2.show()

For extra info my autoexec.be when I used my code:
import lv_tasmota_log
import lv_tasmota_info
import lv_wifi_graph

import haspmota
haspmota.start ()

Sorry for this long description, and hope this will help

Sorry, I cannot make the comparaison between openhasp config for the screen and display.ini for haspmota.
(do not know where to find display config equivalent on openhasp)

**Reply by Be-Virtual** [reply]
If this can help, arduino definitions for this Sunton 8048S050:
#include <Arduino_GFX_Library.h>
#include "BmpClass.h"
#include "Arduino.h"
#include "Audio.h"
#include "SPI.h"
#include "SD.h"
#include "FS.h"
#include <TAMC_GT911.h>

#define BMP_FILENAME "/logo.bmp"

#define TFT_BL 2

#define SD_SCK 12
#define SD_MISO 13
#define SD_MOSI 11
#define SD_CS 10

#define I2S_DOUT 17
#define I2S_BCLK 19
#define I2S_LRC 18

#define TOUCH_SCL 20
#define TOUCH_SDA 19
#define TOUCH_INT -1
#define TOUCH_RST 38
#define TOUCH_ROTATION ROTATION_NORMAL

#define TOUCH_MAP_X1 0
#define TOUCH_MAP_X2 800
#define TOUCH_MAP_Y1 480
#define TOUCH_MAP_Y2 0 

TAMC_GT911 ts = TAMC_GT911(TOUCH_SDA, TOUCH_SCL, TOUCH_INT, TOUCH_RST, max(TOUCH_MAP_X1, TOUCH_MAP_X2), max(TOUCH_MAP_Y1, TOUCH_MAP_Y2));

**Reply by Be-Virtual** [reply]
Bump

**Comment by Be-Virtual** [comment, marked as answer]
Fixed by #18589
